concordium-std/src/test_infrastructure.rs

```diff
@@ -1822,6 +1824,66 @@ impl<State: StateClone<TestStateApi>> TestHost<State> {
     }
 }
 
+#[cfg(all(feature = "wasm-test", feature = "concordium-quickcheck", target_arch = "wasm32"))]
+use getrandom::register_custom_getrandom;
+#[cfg(all(feature = "wasm-test", feature = "concordium-quickcheck", target_arch = "wasm32"))]
+/// A custom function for generating random numbers.
+/// On-chain Wasm cannot use RNGs and function redirects a call to the
+/// `get_random()` primitive (host function), which is later handled by
+/// `TestHost`, where the actual random number generation happens.
+fn get_random(dest: &mut [u8]) -> Result<(), getrandom::Error> {
+    unsafe {
+        crate::prims::get_random(dest.as_mut_ptr(), dest.len() as u32);
+    }
+    Ok(())
+}
+
+#[cfg(all(feature = "wasm-test", feature = "concordium-quickcheck", target_arch = "wasm32"))]
+// Register our own custom random number generation function, so all the calls,
+// that depend on `getrandom` (like `from_entropy()`) will call our function
+// instead. This is only relevant for Wasm, where RNGs are not
+// supported.
```

Comment:
The last sentence is a bit misleading. The same sort of thing actually happens in, e.g., x86 code. Typically to get randomness you will call a kernel provided function. So it is really no different.

Reply:
Fixed.